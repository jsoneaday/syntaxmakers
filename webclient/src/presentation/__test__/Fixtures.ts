import DevProfile from "../models/DevProfile";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";
import { faker } from "@faker-js/faker";
import { getFakeFullName } from "../../domain/TestUtils";

export const defaultDevProfile: DevProfile = {
  key: uuidv4(),
  id: "1",
  updatedAt: new Date().toISOString(),
  userName: "ronaldo", //faker.internet.userName(),
  fullName: getFakeFullName(),
  email: faker.internet.email(),
  primaryLangId: "1",
};
