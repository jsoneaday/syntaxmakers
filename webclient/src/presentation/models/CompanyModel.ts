import { Key } from "react";
import { Company } from "../../domain/repository/CompanyRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type CompanyModel = {
  key: Key;
  id: string;
  updatedAt: string;
  name: string;
};
export default CompanyModel;

export function convert(company: Company): CompanyModel {
  return {
    key: uuidv4(),
    id: company.id,
    updatedAt: company.updatedAt,
    name: company.name,
  };
}
