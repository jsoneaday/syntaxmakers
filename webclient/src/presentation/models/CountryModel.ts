import { Key } from "react";
import { Country } from "../../domain/repository/CountryRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type CountryModel = {
  key: Key;
  id: string;
  updatedAt: string;
  name: string;
};
export default CountryModel;

export function convert(country: Country): CountryModel {
  return {
    key: uuidv4(),
    id: country.id,
    updatedAt: country.updatedAt,
    name: country.name,
  };
}
