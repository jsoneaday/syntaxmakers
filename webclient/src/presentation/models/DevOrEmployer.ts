export enum DevOrEmployer {
  Developer = "Developer",
  Employer = "Employer",
}
