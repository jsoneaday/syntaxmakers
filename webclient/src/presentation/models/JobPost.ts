import { Key } from "react";
import { KeyItem } from "../common/utils";

export default class JobPost implements KeyItem {
  constructor(
    public key: Key,
    public id: bigint,
    public updatedAt: string,
    public title: string,
    public description: string,
    public companyName: string,
    public isRemote: boolean,
    public primaryLangName: string,
    public secondaryLangName: string,
    public industryName: String,
    public salary: string,
    public logo?: Blob,
    public countryName?: string
  ) {}
}
