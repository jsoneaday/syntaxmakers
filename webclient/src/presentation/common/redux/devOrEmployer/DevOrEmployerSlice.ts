import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { DevOrEmployer } from "../../../models/DevOrEmployer";

const initialState: DevOrEmployer = DevOrEmployer.Developer;

const devOrEmployerSlice = createSlice({
  name: "devOrEmployer",
  initialState,
  reducers: {
    setDevOrEmployer: (state: any, action: PayloadAction<DevOrEmployer>) => {
      state = action.payload;
      return state;
    },
  },
});

export const { setDevOrEmployer } = devOrEmployerSlice.actions;
export default devOrEmployerSlice.reducer;
