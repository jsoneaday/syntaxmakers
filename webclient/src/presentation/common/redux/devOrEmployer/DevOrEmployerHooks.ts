import { UiDevOrEmployer } from "../../../models/DevOrEmployer";
import { RootState } from "../Store";
import { useAppDispatch, useAppSelector } from "../StoreHooks";
import { setDevOrEmployer as setDevOrEmployerSlice } from "./DevOrEmployerSlice";

export function useDevOrEmployer(): [
  devOrEmployer: UiDevOrEmployer,
  setDevOrEmployer: (devOrEmp: UiDevOrEmployer) => void
] {
  const devOrEmployer = useAppSelector(
    (state: RootState) => state.devOrEmployer
  );
  const dispatch = useAppDispatch();

  const setDevOrEmployer = (devOrEmp: UiDevOrEmployer) => {
    dispatch(setDevOrEmployerSlice(devOrEmp));
  };

  return [devOrEmployer, setDevOrEmployer];
}
