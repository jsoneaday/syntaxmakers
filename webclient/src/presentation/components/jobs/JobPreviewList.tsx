import { RoutePaths } from "../../../App";
import { useDevOrEmployer } from "../../common/redux/devOrEmployer/DevOrEmployerHooks";
import { startViewTransition } from "../../common/transitions/ViewTransition";
import { DevOrEmployer } from "../../models/DevOrEmployer";
import JobPost from "../../models/JobPost";
import Lister from "../Lister";
import JobPreview from "./JobPreview";
import { Link } from "react-router-dom";

interface JobPreviewListProps {
  jobPosts: JobPost[];
}

export default function JobPreviewList({ jobPosts }: JobPreviewListProps) {
  const [devOrEmp] = useDevOrEmployer();

  const onClickSelectJob = () => {
    startViewTransition(() => {});
  };

  return (
    <div className="dev-post-preview-container" style={{ padding: "2em" }}>
      {jobPosts.length === 0 ? (
        <strong>No jobs found</strong>
      ) : (
        <Lister
          dataItems={jobPosts}
          elementCreator={(dataItem) => (
            <li key={dataItem.key} className="dev-preview-item">
              <Link
                to={
                  devOrEmp === DevOrEmployer.Developer
                    ? RoutePaths.DevJobFullView
                    : RoutePaths.EmpJobFullView
                }
                state={dataItem}
                onClick={onClickSelectJob}
              >
                <JobPreview jobPost={dataItem} isSmall={false} />
              </Link>
            </li>
          )}
        />
      )}
    </div>
  );
}
