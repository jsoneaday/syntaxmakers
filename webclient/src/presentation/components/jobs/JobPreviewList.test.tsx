import configureStore from "redux-mock-store";
import { defaultEmpProfile } from "../../__test__/Fixtures";
import { UiDevOrEmployer } from "../../models/DevOrEmployer";
import { render, screen } from "@testing-library/react";
import * as DevOrEmployerHooks from "../../common/redux/devOrEmployer/DevOrEmployerHooks";
import { Provider } from "react-redux";
import { RouterProvider, createBrowserRouter } from "react-router-dom";
import JobPost from "../../models/JobPost";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";
import JobPreviewList from "./JobPreviewList";
import { EMP_ROUTE_PREFIX } from "../../../App";

const title1 = "Senior Super Dev";
const title2 = "French Frie Dev";

const jobPosts: JobPost[] = [];
jobPosts.push({
  key: uuidv4(),
  id: 1,
  updatedAt: "2022/01/03T22:21:02",
  title: title1,
  description: '[{"type":"paragraph","children":[{"text":"Testing A"}]}]',
  employerId: 1,
  employerName: "Employer John",
  companyId: 1,
  companyName: "Company A",
  isRemote: true,
  primaryLangId: 1,
  primaryLangName: "JavaScript",
  secondaryLangId: 1,
  secondaryLangName: "C#",
  industryId: 1,
  industryName: "Finance",
  salaryId: 1,
  salary: "200000",
  countryName: undefined,
});
jobPosts.push({
  key: uuidv4(),
  id: 2,
  updatedAt: "2022/01/03T22:21:02",
  title: title2,
  description: '[{"type":"paragraph","children":[{"text":"Testing A"}]}]',
  employerId: 2,
  employerName: "Employer Linda",
  companyId: 2,
  companyName: "Company B",
  isRemote: false,
  primaryLangId: 2,
  primaryLangName: "Ruby",
  secondaryLangId: 2,
  secondaryLangName: "Python",
  industryId: 2,
  industryName: "Finance",
  salaryId: 2,
  salary: "200000",
  countryName: "United States",
});

const router = createBrowserRouter([
  {
    path: "/",
    element: <JobPreviewList jobPosts={jobPosts} />,
  },
]);

describe("Test JobPreviewList", () => {
  beforeAll(() => {
    jest
      .spyOn(DevOrEmployerHooks, "useDevOrEmployer")
      .mockImplementation(() => {
        return [UiDevOrEmployer.Employer, (_devOrEmp: UiDevOrEmployer) => {}];
      });
  });

  it("matches snapshot", async () => {
    const mockStore = configureStore();
    const store = mockStore({
      profile: defaultEmpProfile,
      devOrEmployer: UiDevOrEmployer.Employer,
    });

    const result = render(
      <Provider store={store}>
        <RouterProvider router={router} />
      </Provider>
    );

    expect(result).toMatchSnapshot();
  });

  it("links go to emphome", async () => {
    const mockStore = configureStore();
    const store = mockStore({
      profile: defaultEmpProfile,
      devOrEmployer: UiDevOrEmployer.Employer,
    });

    render(
      <Provider store={store}>
        <RouterProvider router={router} />
      </Provider>
    );

    expect(screen.getByText(title1).closest("a")).toHaveAttribute(
      "href",
      `/${EMP_ROUTE_PREFIX}/jobfullview`
    );
  });
});
