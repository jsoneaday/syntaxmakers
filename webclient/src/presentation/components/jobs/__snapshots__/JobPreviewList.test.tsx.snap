// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test JobPreviewList matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dev-post-preview-container"
        style="padding: 2em;"
      >
        <ul
          class="lister-container"
        >
          <li
            class="dev-preview-item"
          >
            <a
              href="/emphome/jobfullview"
            >
              <div
                class="post-preview-container"
              >
                <img
                  class="preview-icon"
                  src=""
                />
                <div
                  class="preview-content"
                >
                  <div
                    class="title-font"
                  >
                    Senior Super Dev
                  </div>
                  <div
                    class="sub-title-font"
                  >
                    Company A
                  </div>
                  <div
                    class="normal-font"
                  >
                    Remote
                  </div>
                  <div
                    class="normal-font preview-salary"
                  >
                    Base Salary: 
                    <i>
                      200000
                    </i>
                  </div>
                  <div
                    class="small-font preview-timestamp"
                  >
                    2022/01/03T22:21:02
                  </div>
                </div>
              </div>
            </a>
          </li>
          <li
            class="dev-preview-item"
          >
            <a
              href="/emphome/jobfullview"
            >
              <div
                class="post-preview-container"
              >
                <img
                  class="preview-icon"
                  src=""
                />
                <div
                  class="preview-content"
                >
                  <div
                    class="title-font"
                  >
                    French Frie Dev
                  </div>
                  <div
                    class="sub-title-font"
                  >
                    Company B
                  </div>
                  <div
                    class="normal-font"
                  >
                    United States
                  </div>
                  <div
                    class="normal-font preview-salary"
                  >
                    Base Salary: 
                    <i>
                      200000
                    </i>
                  </div>
                  <div
                    class="small-font preview-timestamp"
                  >
                    2022/01/03T22:21:02
                  </div>
                </div>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dev-post-preview-container"
      style="padding: 2em;"
    >
      <ul
        class="lister-container"
      >
        <li
          class="dev-preview-item"
        >
          <a
            href="/emphome/jobfullview"
          >
            <div
              class="post-preview-container"
            >
              <img
                class="preview-icon"
                src=""
              />
              <div
                class="preview-content"
              >
                <div
                  class="title-font"
                >
                  Senior Super Dev
                </div>
                <div
                  class="sub-title-font"
                >
                  Company A
                </div>
                <div
                  class="normal-font"
                >
                  Remote
                </div>
                <div
                  class="normal-font preview-salary"
                >
                  Base Salary: 
                  <i>
                    200000
                  </i>
                </div>
                <div
                  class="small-font preview-timestamp"
                >
                  2022/01/03T22:21:02
                </div>
              </div>
            </div>
          </a>
        </li>
        <li
          class="dev-preview-item"
        >
          <a
            href="/emphome/jobfullview"
          >
            <div
              class="post-preview-container"
            >
              <img
                class="preview-icon"
                src=""
              />
              <div
                class="preview-content"
              >
                <div
                  class="title-font"
                >
                  French Frie Dev
                </div>
                <div
                  class="sub-title-font"
                >
                  Company B
                </div>
                <div
                  class="normal-font"
                >
                  United States
                </div>
                <div
                  class="normal-font preview-salary"
                >
                  Base Salary: 
                  <i>
                    200000
                  </i>
                </div>
                <div
                  class="small-font preview-timestamp"
                >
                  2022/01/03T22:21:02
                </div>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
